File: lib/ios/ColorParser.m
#import "ColorParser.h"
#import "NullColor.h"
#import "NoColor.h"
#import <React/RCTConvert.h>

@implementation ColorParser

+ (Color *)parse:(NSDictionary *)json key:(NSString *)key {
	if (json[key]) {
		return [[Color alloc] initWithValue:[RCTConvert UIColor:json[key]]] ?: [NoColor new];
	}
	return [NullColor new];
}

@end

File: lib/android/app/src/reactNative63/java/com/reactnativenavigation/options/parsers/ColorParser.java
package com.reactnativenavigation.options.parsers;

import android.content.Context;

import com.facebook.react.bridge.ColorPropConverter;
import com.reactnativenavigation.options.params.Colour;
import com.reactnativenavigation.options.params.DontApplyColour;
import com.reactnativenavigation.options.params.NullColor;
import com.reactnativenavigation.options.parsers.JSONParser;

import org.json.JSONObject;

public class ColorParser {
    public static Colour parse(Context context, JSONObject json, String colorName) {
        if (json.has(colorName)) {
            Object color = json.opt(colorName);
            if (color == null) {
                return new DontApplyColour();
            } else if (color instanceof Integer) {
                return new Colour(json.optInt(colorName));
            }
            Object convertedColor = JSONParser.convert(json.optJSONObject(colorName));
            Integer processedColor = ColorPropConverter.getColor(convertedColor, context);
            if (processedColor != null) {
                return new Colour(processedColor);
            }
        }
        return new NullColor();
    }
}
